build.sbt

```diff
@@ -851,3 +850,154 @@ commands ++= {
 }
 
 addCommandAlias("scalap",   "scalap/compile:runMain              scala.tools.scalap.Main -usejavacp")
+
+lazy val intellij = taskKey[Unit]("Update the library classpaths in the IntelliJ project files.")
+
+def moduleDeps(name: String) = Def.taskDyn { Def.task((name, (externalDependencyClasspath in Compile in LocalProject(name)).value.map(_.data))) }
+
+intellij := {
+  import xml._
+  import xml.transform._
+
+  val s = streams.value
+
+  val modules: List[(String, Seq[File])] = {
+    // for the sbt build module, the dependencies are fetched from the project's build using sbt-buildinfo
+    val buildModule = ("scala-build", scalabuild.BuildInfo.buildClasspath.split(":").toSeq.map(new File(_)))
+    // `sbt projects` lists all modules in the build
+    buildModule :: List(
+      moduleDeps("actors").value,
+      moduleDeps("compiler").value,
+      // moduleDeps("dist").value,                // No sources, therefore no module in IntelliJ
+      moduleDeps("forkjoin").value,
+      moduleDeps("interactive").value,
+      moduleDeps("junit").value,
+      moduleDeps("library").value,
+      // moduleDeps("library-all").value,         // No sources
+      moduleDeps("manual").value,
+      moduleDeps("partest-extras").value,
+      moduleDeps("partest-javaagent").value,
+      moduleDeps("reflect").value,
+      moduleDeps("repl").value,
+      moduleDeps("repl-jline").value,
+      // moduleDeps("repl-jline-embedded").value, // No sources
+      // moduleDeps("root").value,                // No sources
+      // moduleDeps("scala-dist").value,          // No sources
+      moduleDeps("scaladoc").value,
+      moduleDeps("scalap").value,
+      moduleDeps("test").value)
+  }
+
+  def moduleDep(name: String, jars: Seq[File]) = {
+    val entries = jars.map(f => s"""        <root url="jar://${f.getCanonicalPath}!/" />""").mkString("\n")
```

Comment:
`f.getCanonicalPath` only works for "simple" paths (e.g. no spaces) and only on Unix systems. `f.toURI.getRawPath` should be more robust.